cmd/sign/main.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2023 The Armored Witness authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// The sign tool signs an input file in the
+// [note](https://pkg.go.dev/golang.org/x/mod/sumdb/note) format with a key
+// from Google Cloud Platform's
+// [Key Management Service](https://cloud.google.com/kms/docs).
+//
+// It is intended to be used to sign a manifest file for the Armored Witness
+// firmware transparency log.
+package main
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"cloud.google.com/go/kms/apiv1"
+	"golang.org/x/mod/sumdb/note"
+
+	"cloud.google.com/go/kms/apiv1/kmspb"
+)
+
+type signer struct {
+	ctx     context.Context
```

Comment:
It's quite unusual for structs to contain a `Context`, might be worth a comment here explaining the situation (I imagine it's because of the `note.Signer` interface?)

Reply:
Done.